### DISCUSSION TITLE
Helpful guidance to users on the IP address

### DISCUSSION THREAD

**Discussion by johnbenac** [erew123/alltalk_tts#113] [General]
it isn't totally clear what the user should set the IP address when they're on a network. you might add some information that they need to change the IP address if they hope to access the API remotely. I got it working. In AllTalk, right in the config, you specify the IP address. You can set it to anything you want, even things that won't work (eg. 'foo'), but in my case, I had to set it to the IP address that the AllTalk machine was assigned on the network, if I wanted it to work on the network. The default of 127.0.0.1 would only work if you were trying to use it on the local machine.

I happen to be using zero tier one as my networking solution, so I can access apps on the road using my android, or any other device in my zero tier one network.

Now that I set AllTalk to the IP address assigned to the machine by Zero Tier One, AllTalk will be responsive when it's API is used remotely, using the right address. In this case, it is another app itself that is calling the AllTalk API to generate audio.

Adding just a few words in the docs about this might help users more quickly get their AllTalk working for remote users.

**Comment by erew123** [comment]
Hi @johnbenac 

Thanks for the suggestion. I've added documentation for this on the main Github page here https://github.com/erew123/alltalk_tts?tab=readme-ov-file#-changing-alltalks-ip-address--accessing-alltalk-over-your-network

And also in the help section https://github.com/erew123/alltalk_tts?tab=readme-ov-file#-i-cannot-access-alltalk-from-another-machine-on-my-network

Hopefully that should cover it for anyone who needs it, but if you have any comment, feel free to reply.

Thanks

**Comment by johnbenac** [comment]
looks great! The only tweak that I might make, and it probably isnt needed,
is to make your advice more general than to say what is on the network
card. I use Zero Tier One, and I have no idea what my network card IP
address is. I have to log into the Zero Ties One website to know what the
ip address is for the AllTalk computer on my zero tier network, through
which I access my all talk instance.

Now, you obviously dont need to go into all that detail, but maybe talk
about how it's the IP address of the AllTalk computer on the network where
you are using it, and not necessarily whats on the network card.

**Comment by aios-ai** [comment]
The link to the tts_chunk_player has the IP hard coded: http://127.0.0.1:7851/static/tts_chunk_player/tts_chunk_player.html and it will not work if you changed the IP to something like 192.168.0.10. It would be good if all IP references in the code would be dynamic.

**Comment by erew123** [comment]
@johnbenac I've slightly amended it, but its too complicated for me to cover off large varieties of OS's, network scenarios, cloud infrastructure etc. I think should be enough to give the basic guidance for people to work their way through any custom scenario they may have. 

@Sascha353 I think you meant on the Admin/settings page. Ive set them to dynamically pull the IP based on the IP that you access the AllTalk Admin/settings page from.
